internal/audit/hub/ingest.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021-2024 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package hub
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/cerbos/cerbos/internal/util"
+	"github.com/cerbos/cloud-api/base"
+	logsv1 "github.com/cerbos/cloud-api/genpb/cerbos/cloud/logs/v1"
+	"github.com/cerbos/cloud-api/logcap"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+)
+
+type ErrIngestBackoff struct {
+	underlying error
+	Backoff    time.Duration
+}
+
+func (e ErrIngestBackoff) Error() string {
+	return e.underlying.Error()
+}
+
+type IngestSyncer interface {
+	Sync(context.Context, *logsv1.IngestBatch) error
+}
+
+type Impl struct {
+	client *logcap.Client
+	log    *zap.Logger
+}
+
+func NewIngestSyncer(conf *Conf, logger *zap.Logger) (*Impl, error) {
+	pdpID := util.PDPIdentifier(conf.Ingest.Credentials.PDPID)
```

Comment:
I think we may need to update the `util.PDPIdentifier` function a bit. If the user has not specified a PDP ID, we generate one. I think that ID changes each time you call the function. So, we should use a `sync.OnceValue` to keep the generated ID consistent for the lifetime of the process.

Reply:
Ah good catch